stream_alert/classifier/payload/payload_base.py

```diff
@@ -0,0 +1,222 @@
+        """
+        self.raw_record = raw_record
+        self.resource = resource
+        self.log_source = None
+        self.records = None
+        self.data_type = None
+        self.fully_classified = True
+
+    def __nonzero__(self):
+        return all([self.resource, self.data_type, self.log_source,
+                    self.records, self.fully_classified])
+
+    # For forward compatibility to Python3
+    __bool__ = __nonzero__
+
+    def __repr__(self):
+        return '<{} valid:{} log_source:{} resource:{} type:{} record:{}>'.format(
+            self.__class__.__name__,
+            bool(self),
+            self.log_source,
+            self.resource,
+            self.data_type,
+            self.records
+        )
+
+    @property
+    def log_type(self):
+        return self.log_source.split(':')[0]
+
+    @property
+    def log_subtype(self):
+        return self.log_source.split(':')[-1]
+
+    @classmethod
+    def load_from_raw_record(cls, raw_record):
+        """Extract the originating AWS service and resource from a raw record
+
+        Each raw record contains a set of keys that represent its source.
+        A Kinesis record will contain a 'kinesis' key, while a S3 record
+        contains 's3', and an SNS record contains an 'Sns' key, and so on
+
+        This method also supports loading an S3 event notification that is received via SNS.
+
+        Args:
+            raw_record (dict): A raw record as a dictionary
+
+        Returns:
+            StreamPayload: Loaded subclass of StreamPayload for the proper payload type
+        """
+        resource_mapper = {
+            'kinesis': lambda r: r['eventSourceARN'].split('/')[-1],
+            's3': lambda r: r['s3']['bucket']['name'],
+            'stream_alert_app': lambda r: r['stream_alert_app']
+        }
+
+        service, resource = None, None
+        # check raw record for either kinesis, s3, or apps keys
+        for svc, map_function in resource_mapper.iteritems():
+            if svc in raw_record:
+                # map the resource name from a record
+                resource = map_function(raw_record)
+                service = svc
+                break
+
+        # Sns is capitalized below because this is how AWS stores it within the Record
+        # Other services above, like s3, are not stored like this. Do not alter it!
+        if 'Sns' in raw_record:
+            # If this is an s3 event notification via SNS, extract the bucket from the record
+            if (raw_record['Sns'].get('Type') == 'Notification' and
+                    raw_record['Sns'].get('Subject') == 'Amazon S3 Notification'):
+                service = 's3'
+                raw_record = json.loads(raw_record['Sns']['Message'])['Records'][0]
+                resource = resource_mapper[service](raw_record)
+            else:
+                service = 'sns'
+                resource = raw_record['Sns']['TopicArn'].split(':')[-1]
```

Comment:
why is this not in the resource_mapper?

Reply:
make a change so the default for sns is still in the mapper